### PR TITLE
Gmmin 54 - acceptance tests of effective indexation

### PR DESCRIPTION
**Pull request by awojciak**

### CODE REVIEW

File: gmm_client/src/http_executor.erl [hunk 1]
```diff
@@ -47,7 +47,7 @@ request(Method, Url, RawBody, GetResponse) ->
             _ -> client_utils:encode(RawBody)
         end,
     ReqHeaders = [{<<"Content-Type">>, <<"application/json">>}],
-    case hackney:request(Method, Url, ReqHeaders, Body) of
+    case hackney:request(Method, Url, ReqHeaders, Body, [{pool, false}]) of
```
Comment (MrPaulMarshall) [inline gmm_client/src/http_executor.erl:50]: What does it do?

File: gmm_client/src/http_utils.erl [hunk 2]
```diff
@@ -33,13 +33,14 @@ get_address(_) ->
 
 -spec build_url(Address :: binary(), Path :: binary()) -> binary().
 build_url(Address, Path) ->
-    << Address/binary, "/", Path/binary >>.
+    binary:replace(<< Address/binary, "/", Path/binary >>, <<" ">>, <<"_">>, [global]).
```
Comment (MrPaulMarshall) [inline gmm_client/src/http_utils.erl:36]: How does this work and why is it useful?

Comment (awojciak) [reply gmm_client/src/http_utils.erl:36]: it's simple - it replaces all spaces with underscores - when there were spaces in urls, errors 505 happened

File: gmm_client/src/http_executor.erl (hunk 1 above)
Comment (awojciak) [reply gmm_client/src/http_executor.erl:50]: it disables thread pools - without this, errors "checkout_timeout" happened when there were more than 40 vertices

File: gmm_client/src/http_utils.erl (hunk 2 above)
Comment (MrPaulMarshall) [reply gmm_client/src/http_utils.erl:36]: fair enough, but were there any such situations in the first place?

Comment (awojciak) [reply gmm_client/src/http_utils.erl:36]: Yes, Kamil's generator generates names of some vertices with spaces

File: gmm_client/src/test_utils.erl [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+-module(test_utils).
+-include_lib("stdlib/include/assert.hrl").
+
+%% API
+-export([load_graph/1, random_operations/3]).
+
+add_vertex(VertexData) ->
+    [Zone, Name] = string:split(maps:get(<<"id">>, VertexData), ":"),
+    ok = client:add_vertex(Zone, maps:get(<<"type">>, VertexData), Name),
+    <<Zone/binary, "/", Name/binary>>.
```
Comment (MrPaulMarshall) [inline gmm_client/src/test_utils.erl:16]:
```suggestion
    <<Zone/binary, ":", Name/binary>>.
```

Comment (MrPaulMarshall) [inline gmm_client/src/test_utils.erl:16]:
```suggestion
    <<Zone/binary, ":", Name/binary>>.
```

File: gmm_client/src/test_utils.erl [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+add_edge(EdgeData) ->
+    [ZoneFrom, NameFrom] = string:split(maps:get(<<"src">>, EdgeData), ":"),
+    [ZoneTo, NameTo] = string:split(maps:get(<<"dst">>, EdgeData), ":"),
+    From = <<ZoneFrom/binary, "/", NameFrom/binary>>,
+    To = <<ZoneTo/binary, "/", NameTo/binary>>,
```
Comment (MrPaulMarshall) [inline gmm_client/src/test_utils.erl:22]:
```suggestion
    From = maps:get(<<"src">>, EdgeData),
    To = maps:get(<<"dst">>, EdgeData),
```

**Review by MrPaulMarshall — CHANGES_REQUESTED** [review]

**Review by MrPaulMarshall — DISMISSED** [review]

**Pull request closed without merging**